newsroom/utils.py

```diff
@@ -670,6 +670,14 @@ def get_groups(company: Optional[Company], groups: List[Group]) -> List[Group]:
     """
     filter agenda groups based on restrict_coverage_info permission
     """
-    if company and company.get("restrict_coverage_info"):
-        return [group for group in groups if "restrict_coverage_info" not in group.get("permissions", [])]
-    return groups
+    return [group for group in groups if has_permissions(group, company)]
+
+
+def has_permissions(group: Group, company: Optional[Company]) -> bool:
+    group_permissions = group.get("permissions", [])
+    if group_permissions:
+        for permission in group_permissions:
+            if company and company.get(permission):
+                return False
```

Comment:
this would return `False` if the company has the permission, it should be the other way around

Reply:
would be ideal to add some unittest for this method

Reply:
so this scenario according to our AC, as we have to hide that filter when 
company `permission[restrict_coverage_info] = true`

Reply:
my bad, I realized now we store it the other way around.
would be good to do something like this for future:
```py
Permissions = Literal["coverage_info"]

def get_company_permissions(company: Company) -> Dict[Permissions, bool]:
    return {"coverage_info": not company or company.get("restrict_coverage_info") != True}

def get_groups(groups, company):
    company_permissions = get_company_permissions(company)
    return [group for group in groups if not group.get("permissions") or all([
        company_permissions[permission] for permission in group["permissions"]
    ])]
```
we might change the way how we store what's enabled for company